mapie/tests/test_quantile_regression.py

```diff
@@ -762,3 +771,28 @@ def test_consistent_class() -> None:
     np.testing.assert_allclose(y_pis_1[:, 0, 0], y_pis_2[:, 0, 0])
     np.testing.assert_allclose(y_pis_1[:, 1, 0], y_pis_2[:, 1, 0])
     np.testing.assert_allclose(y_pred_1, y_pred_2)
+
+
+@pytest.mark.parametrize("strategy", [*STRATEGIES])
+def test_fit_parameters_passing(
+    strategy: str,
+) -> None:
+    """
+    Test passing fit parameters, here early stopping at iteration 3.
+    """
+    mapie = MapieQuantileRegressor(
+        estimator=gb,
+        **STRATEGIES[strategy]
+        )
+
+    mapie.fit(
+        X_train,
+        y_train,
+        X_calib=X_calib,
+        y_calib=y_calib,
+        sample_weight=None,
+        monitor=early_stopping_monitor
+        )
+
+    for estimator in mapie.estimators_:
+        assert estimator.estimators_.shape[0] == 3
```

Comment:
I'm not sure I understand what's being tested here. By construction, we should have `estimator.estimators_.shape[0] == GradientBoostingRegressor.n_estimators` (`n_estimators int, default=100`) which corresponds to the number of estimators.`monitor=early_stopping_monitor` forces the maximum number of estimators, have I understood correctly?

Reply:
What you see here is the usage of early stopping in boosting algorithms of scikit learn (https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html#sklearn.ensemble.GradientBoostingRegressor.fit).
What is usually done in boosting is to look at the variation of a metric on a validation set (and not the training set), and to stop when this metric is not improving for a defined number of iterations, and thus to stop before reaching the maximum number of estimators (which is indeed 100 by default). That limits the risk of overfitting. You also have that mechanism in neural network for instance.
For test purposes, I looked at what scikit-learn does currently, which is to use a dummy early stopping procedure that stops at a defined number of iteration (https://github.com/scikit-learn/scikit-learn/blob/4ce8e19859cb8b2f2bef197ed5b28beea44ee4b4/sklearn/ensemble/tests/test_gradient_boosting.py#L981). Here I chose 3 in the `early_stopping_monitor` function. That's why I test that the number of estimators used in boosting is indeed 3 instead of 100, which means the additional fit parameters have been taken into account.

Reply:
Very clear, thank you for these details which you can add in the test docstring (2 sentences) so that no one forgets.